pkg/services/clustering/cluster_manager.go

```diff
@@ -128,15 +142,55 @@ func (cm *ClusterManager) handleDispatcherTaskStatus(taskStatus *DispatcherTaskS
 		if taskStatus.success {
 			cm.changeAlertingState(m.CLN_ALERT_STATUS_PROCESSING)
 		}
+	case DISPATCHER_TASK_TYPE_CLEANUP:
+		if taskStatus.success {
+			cm.log.Info("Cleanup task completed")
+			cm.changeAlertingStateAndRunType(m.CLN_ALERT_STATUS_READY, m.CLN_ALERT_RUN_TYPE_NORMAL)
+		}
 	default:
 		cm.log.Error("Status received on unsupported task type "+string(taskStatus.taskType),
 			"status", taskStatus.success, "error", taskStatus.errmsg)
 	}
 
 	if !taskStatus.success {
-		cm.log.Error("Failed to dispatch task", "error", taskStatus.errmsg)
-		cm.changeAlertingState(m.CLN_ALERT_STATUS_READY)
+		cm.log.Error("Failed to dispatch/execute task", "error", taskStatus.errmsg)
+		cm.changeAlertingStateAndRunType(m.CLN_ALERT_STATUS_READY, m.CLN_ALERT_RUN_TYPE_NORMAL)
+	}
+}
+
+func (cm *ClusterManager) cleanupScheduler() bool {
+	if cm.alertingState.status != m.CLN_ALERT_STATUS_READY {
```

Comment:
Why will it not be set to false? In the next 30 secs, cleanupScheduler will run again; if state is ready, it will schedule cleanup and then the cleanupCheck will become false.